fix(transform/styled-components): fix spans of quasis when some quasis removed

overlookmotel opened the pull request:
In CSS minification:

Input:

```js
styled.div`color: /* ${c} */ blue`
//         ^^^^^^^^^^    ^^^^^^^^ spans of `quasis[0]` and `quasis[1]`
```

Minified:

```js
styled.div`color:blue`
```

The one remaining quasi needs to have span which covers both original quasis:

```js
styled.div`color: /* ${c} */ blue`
//         ^^^^^^^^^^^^^^^^^^^^^^
```

Dunqing commented:
### Merge activity

* **Jul 13, 6:19 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
fix(transform/styled-components): fix spans of quasis when some quasis removed (#12240)

In CSS minification:

Input:

```js
styled.div`color: /* ${c} */ blue`
//         ^^^^^^^^^^    ^^^^^^^^ spans of `quasis[0]` and `quasis[1]`
```

Minified:

```js
styled.div`color:blue`
```

The one remaining quasi needs to have span which covers both original quasis:

```js
styled.div`color: /* ${c} */ blue`
//         ^^^^^^^^^^^^^^^^^^^^^^
```
